### PR TITLE
add figs and fig trees

### PR DESCRIPTION
**Pull request by juur**
#### Summary
Content "Add figs and fig trees, etc."

#### Purpose of change
Add figs, fig trees and similar changes.

#### Describe the solution
I looked at plums and grapes and adjusted based on them. I've not added spawning of random fig trees, as they are not really native to New England. Future work could possible have them in greenhouses or backyards maybe. Or even inside houses.

#### Describe alternatives you've considered
Not having figs: bad.

#### Testing
Boot up. Map edit a tree. Travel in time. Harvest the tree.

### CODE REVIEW

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 1]
```diff
@@ -1110,6 +1110,64 @@
       ]
     }
   },
+  {
+    "type": "terrain",
+    "id": "t_tree_fig",
+    "name": "fig tree",
+    "looks_like": "t_tree_plum",
+    "description": "An example of 'Ficus Carica' - the common edible fig tree. Not normally native to New England, but seems to be doing okay.",
```
Comment (TheShadowFerret) [inline data/json/furniture_and_terrain/terrain-flora.json:1118]:
```suggestion
    "description": "An example of 'Ficus Carica' - the common edible fig tree.  Not normally native to New England, but seems to be doing okay.",
```

File: data/json/furniture_and_terrain/terrain-flora.json [hunk 2]
```diff
@@ -1110,6 +1110,64 @@
...
+        { "item": "leaves", "count": [ 10, 20 ] }
+      ]
+    }
+  },
+  {
+    "type": "terrain",
+    "id": "t_tree_fig_harvested",
+    "name": "fig tree",
+    "looks_like": "t_tree_plum",
+    "description": "An example of 'Ficus Carica' - the common edible fig tree. Not normally native to New England, but seems to be doing okay. This one appears to have been picked.",
```
Comment (TheShadowFerret) [inline data/json/furniture_and_terrain/terrain-flora.json:1147]:
```suggestion
    "description": "An example of 'Ficus Carica' - the common edible fig tree.  Not normally native to New England, but seems to be doing okay.  This one appears to have been picked.",
```

**Comment by Maleclypse** [comment]
Pretty sure fig trees are already in Xedra Evolved so please migrate them if that’s accurate.

**Comment by juur** [comment]
> Pretty sure fig trees are already in Xedra Evolved so please migrate them if that’s accurate.

Yup, let me have a go later in the week and deprecating/migrating to the base game.

**Review by RenechCDDA — CHANGES_REQUESTED** [review]
Orphaned content should not be committed without very good reason. Please add the fig tree terrain to relevant mapgen or remove them from your PR.

Note that the New England climate seems too cold for a fig crop. http://www.hort.cornell.edu/4hplants/Fruits/Fig.html

**Comment by juur** [comment]
> Orphaned content should not be committed without very good reason. Please add the fig tree terrain to relevant mapgen or remove them from your PR.
> 
> Note that the New England climate seems too cold for a fig crop. http://www.hort.cornell.edu/4hplants/Fruits/Fig.html

The terrain is used by Xendra Evolved (and maybe planting?). Is that OK or should I add it in somewhere?

It is possible for some varieties of this species to grow in the UK, which I think is a little cooler than New England: https://www.rhs.org.uk/fruit/figs/grow-your-own

**Comment by juur** [comment]
> Pretty sure fig trees are already in Xedra Evolved so please migrate them if that’s accurate.

Done, I think.

**Comment by Maleclypse** [comment]
Add them as a potential line of trees grown at a tree farm or something.  They do need to at least spawn somewhere in vanilla to justify adding them to mainline.

**Review by kevingranade — CHANGES_REQUESTED** [review]
> It is possible for some varieties of this species to grow in the UK, which I think is a little cooler than New England: https://www.rhs.org.uk/fruit/figs/grow-your-own

One,  check your source,  it's saying with extra extra special care,  up to moving the tree indoors for the winter,  you can maybe grow figs in some parts of the uk. That doesn't really qualify as working.

Two, simply asserting that the uk is cooler than new england (I have no idea if this is at all true) is insufficient, we're looking for a source that says that it should work in New England.

Three, for plants that have marginal survivability in the area, what should be added is a version of the plant that does not produce fruit without manual intervention.

**Pull request closed without merging**